temporal/server_test.go

```diff
@@ -48,9 +49,31 @@ import (
 // TestNewServer verifies that NewServer doesn't cause any fx errors, and that there are no unexpected error logs after
 // running for a few seconds.
 func TestNewServer(t *testing.T) {
-	t.Parallel()
+	startAndStopServer(t)
+}
 
+// TestNewServerWithOTEL verifies that NewServer doesn't cause any fx errors when OTEL is enabled.
+func TestNewServerWithOTEL(t *testing.T) {
+	t.Setenv("OTEL_TRACES_EXPORTER", "otlp")
+	t.Setenv("OTEL_BSP_SCHEDULE_DELAY", "100")
+	t.Setenv("OTEL_EXPORTER_OTLP_TRACES_INSECURE", "true")
+
+	t.Run("with OTEL Collector running", func(t *testing.T) {
+		otelLogger, err := otellogger.Start(t)
+		require.NoError(t, err)
+		t.Setenv("OTEL_EXPORTER_OTLP_ENDPOINT", otelLogger.Addr())
+		startAndStopServer(t)
+		require.NotEmpty(t, otelLogger.Spans(), "expected at least one OTEL span")
+	})
+
+	t.Run("without OTEL Collector running", func(t *testing.T) {
+		startAndStopServer(t)
+	})
+}
+
+func startAndStopServer(t *testing.T) {
 	cfg := loadConfig(t)
+	cfg.Global.Metrics.Prometheus.ListenAddress = ":0" // prometheus reporter does not shut down in-between test runs
```

Comment:
I tried shutting down the Prometheus reporter, but there is no way to do that as it starts the http server inside a goroutine and has no stop/shutdown callback :-/